c10/core/Scalar.h

```diff
@@ -88,6 +88,17 @@ class C10_API Scalar {
 
   Scalar operator-() const;
 
+  template<typename T>
+  bool equal(T num) const {
```

Comment:
cc. @ezyang to verify if it's ok to add `equal` for `Scalar`.

Reply:
This seems fine.